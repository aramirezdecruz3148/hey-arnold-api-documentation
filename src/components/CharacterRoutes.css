.flex {
  display: flex;
  flex-direction: column;
  margin-left: 3vw;
}

.codeBlock {
  padding-top: 10px;
  height: 20vh;
  overflow: scroll;
  list-style: none;
  width: 75vw;
  background: whitesmoke;
}

.title {
  font-family: 'Luckiest Guy', cursive;
  color: #FFD90F;
  font-size: 3vw;
  margin-bottom: -.5vw;
  text-shadow:
		-.2vw -.2vw 0 #000,
		.2vw -.2vw 0 #000,
		-.2vw .2vw 0 #000,
    .2vw .2vw 0 #000;
}

