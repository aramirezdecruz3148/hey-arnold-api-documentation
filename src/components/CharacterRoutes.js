import React from 'react';
import allCharacters from '../data/characters';
import paginatedCharacters from '../data/characters-pagination';
import nameCharacters from '../data/characters-name';
import randomCharacter from '../data/character-random';
import characterById from '../data/character-id';

export default function CharacterRoutes() {
  return (
    <>
      <h3>Character Routes</h3>
      <section>
        <h4>Get All Characters</h4>
        <h5>https://hey-arnold-api.herokuapp.com/api/v1/characters</h5>
        <div>
          <pre>{JSON.stringify(allCharacters, null, 2)}</pre>
        </div>
        <h5>Pagination</h5>
        <p>
          This route will return a paginated list.
          By default, a list (or <q>page</q>), will return 20 characters.
          To change this number, simply add a <q>perPage</q> query parameter,
          e.g. ?perPage=3. You can move to the next page of results by adding
          the <q>page</q> query parameter, e.g. ?page=2.
        </p>
        <h5>{'https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=3&page=2'}</h5>
        <div>
          <pre>{JSON.stringify(paginatedCharacters, null, 2)}</pre>
        </div>
        <h5>Query by Character Name</h5>
        <p>
          This route does allow you to query characters by name, with
          the ability to return all partial matches regardless of casing,
          e.g. ?name=hel
        </p>
        <h5>https://hey-arnold-api.herokuapp.com/api/v1/characters?name=hel</h5>
        <div>
          <pre>{JSON.stringify(nameCharacters, null, 2)}</pre>
        </div>
      </section>
      <section>
        <h4>Get Random Character</h4>
        <p>
          This route returns one random character by default. To change this number, 
          simply add a <q>count</q> query parameter, e.g. ?count=2.
        </p>
        <h5>https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2</h5>
        <div>
          <pre>{JSON.stringify(randomCharacter, null, 2)}</pre>
        </div>
      </section>
      <section>
        <h4>Get Character by Id</h4>
        <h5>https://hey-arnold-api.herokuapp.com/api/v1/characters/5da237699734fdcb7bef8f51</h5>
        <div>
          <pre>{JSON.stringify(characterById, null, 2)}</pre>
        </div>
      </section>
    </>
  );
}

