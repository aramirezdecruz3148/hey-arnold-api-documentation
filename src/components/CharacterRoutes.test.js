import React from 'react';
import { shallow } from 'enzyme';
import CharacterRoutes from './CharacterRoutes';

describe('CharacterRoutes component', () => {
  it('renders CharacterRoutes', () => {
    const wrapper = shallow(<CharacterRoutes />);
    expect(wrapper).toMatchSnapshot();
  });
});
