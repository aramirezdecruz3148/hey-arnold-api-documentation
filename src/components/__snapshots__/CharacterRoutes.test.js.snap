// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterRoutes component renders CharacterRoutes 1`] = `
<div
  className="flex"
>
  <h3
    className="title"
  >
    Character Routes
  </h3>
  <section>
    <h4
      className="queries"
    >
      Get All Characters
    </h4>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/characters
    </h5>
    <p
      className="description"
    >
      This route will return a paginated list. By default, a list (or 
      <q>
        page
      </q>
      ), will return 20 characters. To change this number, simply add a 
      <q>
        perPage
      </q>
       query parameter, e.g. ?perPage=3. You can move to the next page of results by adding the 
      <q>
        page
      </q>
       query parameter, e.g. ?page=2.
    </p>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=3&page=2
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        [
  {
    "_id": "5da237699734fdcb7bef8f56",
    "name": "Miles Shortman",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
  },
  {
    "_id": "5da237699734fdcb7bef8f54",
    "name": "Mitzi",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
  },
  {
    "_id": "5da237699734fdcb7bef8f55",
    "name": "Arnie",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
  }
]
      </pre>
    </div>
    <h4
      className="queries"
    >
      Query by Character Name
    </h4>
    <p
      className="description"
    >
      This route does allow you to query characters by name, with the ability to return all partial matches regardless of casing, e.g. ?name=hel.
    </p>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/characters?name=hel
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        [
  {
    "_id": "5da237699734fdcb7bef8f63",
    "name": "Helga G. Pataki",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
  },
  {
    "_id": "5da237699734fdcb7bef8f8b",
    "name": "Shelley",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
  },
  {
    "_id": "5da237699734fdcb7bef902e",
    "name": "Helga's Parrot (bird)",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/f/f1/Helga%27s_parrot%2C_bird.jpg/revision/latest/scale-to-width-down/310?cb=20100307071524"
  },
  {
    "_id": "5da237699734fdcb7bef8fb8",
    "name": "Shelley",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
  }
]
      </pre>
    </div>
  </section>
  <section>
    <h4
      className="queries"
    >
      Get Random Character
    </h4>
    <p
      className="description"
    >
      This route returns one random character by default. To change this number, simply add a 
      <q>
        count
      </q>
       query parameter, e.g. ?count=2.
    </p>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/characters/random?count=2
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        [
  {
    "_id": "5da237699734fdcb7bef9036",
    "name": "Sidney",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/d/d4/Sidney.jpg/revision/latest/scale-to-width-down/310?cb=20120807185330"
  },
  {
    "_id": "5da237699734fdcb7bef8f9e",
    "name": "Edmund",
    "image": "https://vignette.wikia.nocookie.net/heyarnold/images/0/0a/Edmund.png/revision/latest?cb=20171219083557"
  }
]
      </pre>
    </div>
  </section>
  <section>
    <h4
      className="queries"
    >
      Get Character by Id
    </h4>
    <h5
      className="endpoints"
    >
      https://hey-arnold-api.herokuapp.com/api/v1/characters/5da237699734fdcb7bef8f51
    </h5>
    <div
      className="codeBlock"
    >
      <pre>
        {
  "_id": "5da237699734fdcb7bef8f51",
  "name": "Arnold Shortman",
  "image": "https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
}
      </pre>
    </div>
  </section>
</div>
`;
